src/helpers/amoc_xmpp_inbox.erl

```diff
@@ -0,0 +1,69 @@
+-module(amoc_xmpp_inbox).
+
+-include_lib("exml/include/exml.hrl").
+
+-define(V(X), (fun amoc_config_validation:X/1)).
+
+-required_variable(
+   [#{name => max_items_per_inbox_lookup, default_value => 10, verification => ?V(nonnegative_integer),
+      description => "Maximum number of messages returned per inbox lookup"},
+    #{name => inbox_lookup_timeout, default_value => 10, verification => ?V(positive_integer),
+      description => "Timeout for each inbox lookup request (in seconds)"}
+   ]).
+
+-export([init/0, lookup/1, received_handler_spec/0]).
+
+-spec init() -> ok.
+init() ->
+    amoc_xmpp_presence:init(),
+    amoc_metrics:init(counters, inbox_lookups),
+    amoc_metrics:init(counters, inbox_messages_received),
+    amoc_metrics:init(counters, timeouts),
+    amoc_metrics:init(times, inbox_lookup_response_time),
+    ok.
+
+-spec lookup(escalus:client()) -> exml:element().
+lookup(Client) ->
+    Req = escalus_stanza:iq(<<"set">>,
+                            [#xmlel{name = <<"inbox">>,
+                                    attrs = [{<<"xmlns">>, ns(inbox)}],
+                                    children = [rsm_max()]}]),
+    Pred = fun(Stanza) -> escalus_pred:is_iq_result(Req, Stanza) end,
+    Metric = inbox_lookup_response_time,
+    Timeout = cfg(inbox_lookup_timeout),
+    amoc_metrics:update_counter(inbox_lookups),
+    amoc_xmpp:send_request_and_get_response(Client, Req, Pred, Metric, Timeout).
+
+rsm_max() ->
+    Max = cfg(max_items_per_lookup),
```

Comment:
Shouldn't be here `max_items_per_inbox_lookup`?

Reply:
Yes it should be. Good catch! I will fix in a new PR.